cdap-common/bin/cdap

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+#
+# Copyright © 2016 Cask Data, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+#
+# This script is the main script for performing all CDAP functions. With it,
+# you can start, stop, or get status on CDAP and its services, run queries,
+# execute CLI commands, and more. Anything not handled by this script is passed
+# to the CDAP CLI to be interpreted. 
+#
+
+__script=${BASH_SOURCE[0]} __script_bin=$(dirname ${__script})
+__comp_home=${__script%/*/*}
+if [[ ${__comp_home%/*} == /opt/cdap ]]; then
+  __app_home=${__comp_home}
```

Comment:
suggestion: seems like ``__app_home`` is unecessary.  it could simplify things by removing it from here, and from the ``cdap_home`` function, and then just sourcing ``${CDAP_HOME}/bin/functions.sh`` in line 37 below